### PR TITLE
Address javac compiler warnings and intellij inspections

### LINKED ISSUE #250: BugsnagReactNative.java uses unchecked or unsafe operations
### Description
I followed native installation instructions for both iOS and Android. Works fine for iOS, but when building for Android via Android Studio I get the compile error below.

`.../BugsnagReactNative.java: uses unchecked or unsafe operations.`

Full error output is below, under _Error messages:_

### Environment

Library versions:

```shell
warning Resolution field "core-js@2.5.2" is incompatible with requested version "core-js@^1.0.0"
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ bugsnag-react-native@2.9.5
├─ react-native-code-push@5.3.2
└─ react-native@0.52.2
```

- Android version: 7.0 
- Physical device: Samsung Galaxy S7 Edge
- Debug mode
- `BugsnagReactNative.start(this)` is present in the
  `onCreate` method of my `MainApplication` class

**Specifying Google’s maven repository in android/build.gradle as:**
```
allprojects {
    repositories {
        maven { url "https://jitpack.io" }
        maven { url "https://mobile-sdk.jumio.com" }
        mavenLocal()
        jcenter()
        google()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        maven {
            url "https://maven.google.com"
        }
    }
}

```

**Error messages:**
Error when building from Android Studio
```
/Users/.../.../node_modules/bugsnag-react-native/android/src/main/java/com/bugsnag/BugsnagReactNative.java:
uses unchecked or unsafe operations.
Recompile with -Xlint:unchecked for details.
```

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Resolve any compiler warnings by addressing the root cause, or suppressing the warning where that is not feasible.

## Changeset

- Avoid use of static import
- Send correct number of arguments to `String.format`
- Add generic arguments to raw map variables
- Avoid redundant string array instantiation
- Add missing non-null annotation
- Add serialVersionUID to serialisable custom exception
- Suppress type erasure/3rd party warnings

### Linked issues

Fixes #250 

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: android/src/main/java/com/bugsnag/BugsnagReactNative.java [hunk 1]
```diff
@@ -364,6 +369,7 @@ Severity parseSeverity(String value) {
         return output;
     }
 
+    @SuppressWarnings("unchecked") // ignore type erasure when casting Map
```
Comment (kattrali) [inline android/src/main/java/com/bugsnag/BugsnagReactNative.java]: Can this be scoped a bit smaller, e.g.:

```java
@SuppressWarnings("unchecked")
Map<String, Object> values = (Map<String, Object>) value;
```

Comment (fractalwrench) [reply android/src/main/java/com/bugsnag/BugsnagReactNative.java]: Reduced the suppression to local variable scope.
